robottelo/cli/factory.py

```diff
@@ -109,6 +114,226 @@ def create_object(cli_object, options, values):
 
 
 @cacheable
+def activationkey_subscribe_to_custom_repo(options=None):
```

Comment:
the name is misleading and is not conveying what this function is trying to do - May be we can call this something like:
```
setup_org_for_a_custom_repo(options=None)
```

Also the function's docstring should explain what this function is trying to do, something like:
```
"""
Sets up Org for the given custom repo by:
1. Checks for an organization-id else it creates a new org
2. Creates a new product with the custom repo.  Synchronizes the repo.
3. Checks for a content-view-id or creates a new Content view and 
    - adds the custom repo
    - publishes
    - promotes  to the activation key's lifecycle environment
4. Adds the custom repo subscription to the Activation key